### PR TITLE
CHORE: add license headers

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull-request configures the [ansys pre-commit-hooks](https://github.com/ansys/pre-commit-hooks). It also enables the license header checks. Related with #4775.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -64,6 +64,13 @@ repos:
   hooks:
   -   id: check-pre-commit-ci-config
 
+- repo: https://github.com/ansys/pre-commit-hooks
+  rev: v0.3.1
+  hooks:
+    - id: add-license-headers
+      files: '(pyaedt|examples|_unittest|_unittest_ironpython|_unittest_solvers)/.*\.(py)'
+      args:
+      - --start_year=2021
```
Comment (jorgepiloto) [inline .pre-commit-config.yaml:73]: I imposed this year since it is the one for the first public release of PyAEDT.

**Review by Samuelopez-ansys — APPROVED** [review]
Hopefully you did it with a script :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MaxJPRey) [reply .pre-commit-config.yaml:73]: @jorgepiloto Yes all good.

**Comment by jorgepiloto** [comment]
Is it normal that the IronPython tests fail? Seems that some LOG file is missing.

**Comment by jorgepiloto** [comment]
Hi @Samuelopez-ansys, thank's for pointing this out. Let me fix this. I hope the ansys-license-headers hooks does not remove those UTF8 lines.

**Comment by jorgepiloto** [comment]
Changing this to draft. Pinging @klmcadams.

The license headers get placed on top of the `# -*- coding: utf-8 -*-` line. This makes Python to ignore the actual encoding.

Adding a custom template leads to a duplicated `# -*- coding: utf-8 -*-`:

```
# -*- coding: utf-8 -*-
... LICENSE STUFF ...

# -*- coding: utf-8 -*- (the original one)
```

**Comment by jorgepiloto** [comment]
My fault. I accidentally closed this.

**Comment by Samuelopez-ansys** [comment]
@jorgepiloto Could we merge this PR?

**Comment by jorgepiloto** [comment]
I crafted a custom template to ensure the `UTF-8` line is added before every license check. Before adding the headers, I removed every `UTF-8` line using a custom script.

Everything should be ready to go now.

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, thanks for this changes !

**Pull request merged**